addons/mail/models/mail_notification.py

```diff
@@ -113,10 +113,13 @@ def format_failure_reason(self):
 
     def _filtered_for_web_client(self):
         """Returns only the notifications to show on the web client."""
-        return self.filtered(lambda n:
-            n.notification_type != 'inbox' and
-            (n.notification_status in ['bounce', 'exception', 'canceled'] or n.res_partner_id.partner_share)
-        )
+        def filter_notif(notif):
+            if notif.notification_status in ['bounce', 'exception', 'canceled'] or notif.res_partner_id.partner_share:
```

Comment:
can you split this line ?
```suggestion
            if notif.notification_status in ['bounce', 'exception', 'canceled'] \
                or notif.res_partner_id.partner_share:
```